builder/main.py

```
@@ -650,6 +650,70 @@ def firmware_metrics(target, source, env):
+        if cli_args:
+            cmd.extend(cli_args)
+        else:
+            # Default arguments if none provided
+            cmd.extend(["info_corefile", "--chip", mcu, "--elf", elf_file])
+
+        # Debug-Info if wanted
+        if env.GetProjectOption("custom_esp_coredump_verbose", False):
+            print(f"Running command: {' '.join(cmd)}")
+        
+        # Execute esp-coredump with current environment
+        result = subprocess.run(cmd, check=False, capture_output=False, env=os.environ)
```

Ruff (0.13.1)

`subprocess` call: check for execution of untrusted input

(S603)